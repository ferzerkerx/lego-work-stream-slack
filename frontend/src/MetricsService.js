import { Utils } from './Utils';

class MetricsService {
  static createConfiguration(form) {
    const startDate = form.startDate.value || Utils.toDateString(new Date());
    const endDate =
      form.endDate.value || Utils.toDateString(Utils.addDays(startDate, 1));
    const frequency = Utils.toNumber(form.frequency.value) || 1;
    const isPercentage = form.isPercentage.checked || false;
    const teams = Utils.toArray(form.teams.value) || [];

    return {
      startDate: startDate,
      endDate: endDate,
      frequency: frequency,
      isPercentage: isPercentage,
      teams: teams,
    };
  }

  static createUrl(config) {
    return `/api/metrics?startDate=${config.startDate}&endDate=${
      config.endDate
    }&frequency=${config.frequency}&isPercentage=${
      config.isPercentage
    }&teams=${config.teams.join(',')}&format=${config.format}`;
  }
}

export { MetricsService };
