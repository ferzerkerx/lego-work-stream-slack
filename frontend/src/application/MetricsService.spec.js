import { MetricsService } from './MetricsService';

const fakeForm = () => {
  return {
    startDate: { value: '2018-10-01' },
    endDate: { value: '2018-10-01' },
    frequency: { value: 10 },
    isPercentage: { checked: true },
    teams: { value: 'team1,team2' },
  };
};

describe('MetricsService', () => {
  test('createConfiguration', () => {
    const configuration = MetricsService.createConfiguration(fakeForm());
    expect(configuration).toMatchSnapshot();
  });

  test('createUrl', () => {
    const configuration = MetricsService.createConfiguration(fakeForm());
    configuration.format = 'json';
    const url = MetricsService.createUrl(configuration);
    expect(url).toMatchSnapshot();
  });
});
