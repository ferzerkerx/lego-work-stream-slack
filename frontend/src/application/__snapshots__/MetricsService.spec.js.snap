// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricsService createConfiguration 1`] = `
Object {
  "endDate": "2018-10-01",
  "frequency": 10,
  "isPercentage": true,
  "startDate": "2018-10-01",
  "teams": Array [
    "team1",
    "team2",
  ],
}
`;

exports[`MetricsService createUrl 1`] = `"/api/metrics?startDate=2018-10-01&endDate=2018-10-01&frequency=10&isPercentage=true&teams=team1,team2&format=json"`;
